app/models/spree/calculator/flowcommerce_calculator.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module Spree
+  class Calculator
+    class FlowcommerceCalculator < Calculator::DefaultTax
+      def self.description
+        'FlowCommerce Calculator'
+      end
+
+      def compute_shipment_or_line_item(item)
+        order = item.order
+
+        if can_calculate_tax?(order)
+          flow_response = get_flow_response(order)
+          tax_for_item(item, flow_response)
+        else
+          prev_tax_amount(item)
+        end
+      end
+      alias compute_shipment compute_shipment_or_line_item
+      alias compute_line_item compute_shipment_or_line_item
+
+      private
+
+      def prev_tax_amount(item)
+        if rate.included_in_price
+          item.included_tax_total
+        else
+          item.additional_tax_total
+        end
+      end
+
+      def can_calculate_tax?(order)
+        return false if order.flow_data.blank?
+
+        true
+      end
+
+      def get_flow_response(order)
+        flow_io_tax_response = Rails.cache.fetch(order.flow_tax_cache_key, time_to_idle: 5.minutes) do
+          FlowcommerceSpree::Api.run :get, "/:organization/orders/allocations/#{order.number}"
```

Comment:
Better use `flocommerce` gem:

```suggestion
          FlowcommerceSpree.client.orders.get_allocations_by_number(FlowcommerceSpree::ORGANIZATION, order.number) 
```

Reply:
:open_mouth: Nice!! I will try that and if it works ok I will update specs and push the change (if needed)! Thanks for the suggestion!!

Reply:
Done! Had to update the rest of the code (since the response was slightly different) and also the specs!